### DISCUSSION TITLE
intercepting responses in middleware and modifying them

### DISCUSSION THREAD

**Discussion by tx46** [vercel/next.js#57945] [Help] [answered]
### Summary

I have a section of my api (e.g., /api/objects/[type] and similar) that always return json. the json needs to be reprocessed before being sent to the client. I don't want to do this in every single route. instead, I thought using middleware was a good idea.

I do not understand how I can intercept responses in middleware (or if that is even the correct use case for middleware), rewrite the response and then send the modified response back to the client. 

how do I do it?

**Comment by icyJoseph** [comment, marked as answer]
That's not possible with the Next.js middleware.

See more here: https://github.com/vercel/next.js/discussions/49086#discussioncomment-5788198
